doc/source/development/rfc/rfc99_geometry_coordinate_precision.rst

```diff
@@ -0,0 +1,448 @@
+in the files it generates, by adding ``xy_coordinate_resolution`` and
+``z_coordinate_resolution`` members at the FeatureCollection level.
+
+The existing COORDINATE_PRECISION layer creation option, if specified, will
+take precedence over the settings coming from OGRGeomFieldDefn::GetCoordinatePrecision().
+
+GeoJSONSeq
+++++++++++
+
+The driver will compute the number of decimal digits after the decimal point
+to write as ``ceil(1. / resolution)``
+
+It will *not* be able to store it in its metadata.
+
+JSONFG
+++++++
+
+Similar to GeoJSON. One subtelty is that this driver may write both the "place"
+geometry (generally in a non-WGS84 CRS) and the GeoJSON RFC7946 WGS84 "geometry".
+
+The OGRGeomFieldDefn::GetCoordinatePrecision() will qualify the "place" geometry.
+The coordinate precision of the WGS84 "geometry" will be derived from the one
+of the "place" geometry with appropriate geographic/projected CRS and axis unit
+changes.
+
+The coordinate precision metadata of the "place" member will be stored in
+``xy_coordinate_resolution_place`` and ``z_coordinate_resolution_place``
+members at the FeatureCollection level.
+
+For the "geometry" member, the same ``xy_coordinate_resolution`` and
+``z_coordinate_resolution`` members as the GeoJSON driver will be used.
+
+The existing COORDINATE_PRECISION_PLACE or COORDINATE_PRECISION_GEOMETRY layer
+creation option, if specified, will take precedence over the settings coming
+from OGRGeomFieldDefn::GetCoordinatePrecision().
+
+GML
++++
+
+The driver will compute the number of decimal digits after the decimal point
+to write as ``ceil(1. / resolution)``
+
+The driver will be able to store the coordinate precision metadata in the XML
+schema it generates by adding a ``xs:annotation/xs:appinfo`` element in the
+declaration of the geometry property, and with ``ogr:xy_coordinate_resolution``,
+``ogr:z_coordinate_resolution`` and ``ogr:m_coordinate_resolution`` sub-elements.
+This should hopefully be ignored by readers that don't recognize
+that metadata (this will be the case of GDAL < 3.9)
+
+.. code-block:: xml
+
+        <xs:element name="wkb_geometry" type="gml:SurfacePropertyType" nillable="true" minOccurs="0" maxOccurs="1">
+            <xs:annotation>
+              <xs:appinfo source="http://ogr.maptools.org/">
+                <ogr:xy_coordinate_resolution>8.9e-9</ogr:xy_coordinate_resolution>
+                <ogr:z_coordinate_resolution>1e-3</ogr:z_coordinate_resolution>
+                <ogr:m_coordinate_resolution>1e-3</ogr:m_coordinate_resolution>
+              </xs:appinfo>
+            </xs:annotation>
+        </xs:element>
+
+CSV
++++
+
+The driver will compute the number of decimal digits after the decimal point
+to write as ``ceil(1. / resolution)``
+
+It will *not* be able to store it in its metadata. The possibility of storing
+the coordinate metadata in the .csvt side-car file has been considered, but it
+would not be backwards-compatible.
```

Comment:
Can/should precision be detected in the way that column types are?

Reply:
Obviously it *could*, but I'm not sure I want to do that implementation effort :-). It could be relatively easy when constructing point geometries from X,Y columns, but it becomes trickier when importing them from WKT, because our WKT import code should take note of that.